resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileStore.java

```diff
@@ -32,11 +32,12 @@ public class NullFileStore extends FileStore {
 
 	/**
 	 * Creates a null file store
+	 * 
 	 * @param path The path of the file in this store
 	 */
 	public NullFileStore(IPath path) {
 		Assert.isNotNull(path);
-		this.path = path;
+		this.path = path.makeAbsolute();
```

Comment:
```suggestion
		Assert.isLegal(path.isAbsolute());
```

Reply:
It is a null file system so you only reduce its input domain for no reason whatsoever.

Reply:
Passing a non absolute path seems illegal for this (internal) constructor, so it should be asserted but not fix caller errors automatically.